### ISSUE TITLE
Programs with circles do not work on ShopBot

### ISSUE THREAD

**Issue by mikemag** [mikemag/MonkeyCAM#1]
ShopBot doesn't support the R variant of G02, so the following programs fail on ShopBot:
- Inserts
- Guide holes

Need to move to the I,J variant of G02 to support less capable controllers.

**Comment by splitn2** [comment]
@mikemag , along with Shopbot we should look at making the G-code work on GRBL Arduino based cnc too, I've run some MonkeyCAM code through the GRBL test setup I have and it barf'd on one setup command for dynamic tool length offset G43.1, but otherwise seems to work on the bench test, I'll carry on building up my large MACH3 based cnc for board building but will also use this Arduino based GRBL project for a smaller machine. With the right sized steppers and drivers GRBL could easily do the job for a ski/board builder and might help simplify and lower the cost for a CNC for entry level users and given its open source its got a lot of potential to develop into a pretty useful workshop CNC. Would it be best to add a switch in the machine-defs file to select the type of G-code controller MACH3/SHOPBOT/GRBL or would it be easier to dumb it down to the most generic and transportable? Cheers, Rich
